qutebrowser/utils/urlutils.py

```diff
@@ -443,3 +443,60 @@ def __str__(self):
             return self.msg
         else:
             return '{}: {}'.format(self.msg, self.url.errorString())
+
+
+class IncDecError(Exception):
+
+    """Exception raised by url_incdec_number on problems.
+
+    Attributes:
+        msg: The error message.
+        url: The QUrl which caused the error.
+    """
+
+    def __init__(self, msg, url):
+        super().__init__(msg)
+        self.url = url
+        self.msg = msg
+
+    def __str__(self):
+        return '{}: {}'.format(self.msg, self.url.toString())
+
+
+def url_incdec_number(url, incdec):
+    """Find a number in the url and increment or decrement it.
+
+    Args:
+        url: The current url
+        incdec: Either 'increment' or 'decrement'
+
+    Return:
+        The new url with the number incremented/decremented.
+
+    Raises IncDecError if the url contains no number.
+    """
```

Comment:
Please add something like this (and a test for it):

```python
    if not url.isValid():
        raise ValueError(get_errstring(url))
```

This isn't strictly necessary (as it's already enforced by `_current_url()`), but since it's now part of `urlutils` and might be used in other places, let's add it.